libflux/src/flux/semantic/types.rs

```diff
@@ -356,11 +356,13 @@ impl MonoType {
             },
             MonoType::Uint => match with {
                 Kind::Addable
+                | Kind::Subtractable
                 | Kind::Divisible
                 | Kind::Numeric
                 | Kind::Comparable
                 | Kind::Equatable
-                | Kind::Nullable => Ok(Substitution::empty()),
+                | Kind::Nullable
+                | Kind::Signed => Ok(Substitution::empty()),
```

Comment:
It's pretty counter-intuitive to have `MonoType::Uint` be constrainable to `Kind::Signed`, since the "U" in `Uint` stands for "unsigned."  What happens if `Kind::Signed` doesn't appear here?

Reply:
I also thought this. @nathanielc and I looked it, and it turns out that it is still _kinda_ signed, because unary operations. `b = -a` looks like it's signed.

Here's the relevant inference:  https://github.com/rockstar/flux/blob/subtractable-uint/libflux/src/flux/semantic/nodes.rs#L1338

We tried to change the `Kind::Signed` to `Kind::Numeric` (because it felt awkward), but `Kind::Numeric` is also implemented by `Duration`, so it got more complex.

Reply:
Maybe we should choose a new name here? Maybe Negatable? As an analog to Subtractable but for the unary case? The Signed constraint here is used purely to indicate that unary `+` and `-` operations can be applied and has no other application. 

As Paul mentions we need a new constraint beyond Subtractble and Numeric because you can apply the `-` and `+` unary operators to durations but you cannot subtract durations and they are not numeric.

Reply:
Also what ever reasoning we decide should be captured in a comment on the definition of the kind constraint as it took Paul and I a bit to figure out the semantics of the Signed constraint based off its name alone.

Reply:
If `Kind::Signed` is only used for unary arithmetic operators, I wonder if we just need to remove the `Signed` constraint altogether?

Reply:
Or I guess that we need some way of enforcing that you can't do things like `-"foo"`.  What about `Numeric`?  I know that Duration also implements it, but it seems like we want to allow things like `+1h` or `-1d`?

Reply:
Correct we want to allow `+1`, `-1`, `+1h` and `-1h` but we want to disallow `-"foo"` or `-true`. Since Durations are not numeric we need something else to show that unary negation can be applied. Currently that is called `signed`.

Reply:
So should we change what the name is? Or create a new `Kind`? I'm unsure about the action that needs to be taken here.

Reply:
Ok, I would be fine with renaming `Kind::Signed` to `Kind::Negatable`.  Signed means something pretty specific when it comes to types, so I'd just like to avoid confusion.

Reply:
I would say just change the name of the kind, so it reflects that the type can be used in unary arithmetic, and otherwise your changes are good.

Reply:
We could also just call the kind Unary to be really specific. I'm fine with either though.